pkg/servicemesh/federation/imports/controller.go

```diff
@@ -0,0 +1,191 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	"context"
+	"fmt"
+
+	xnsinformers "github.com/maistra/xns-informer/pkg/informers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	maistrainformers "maistra.io/api/client/informers/externalversions/core/v1alpha1"
+	maistraclient "maistra.io/api/client/versioned"
+	"maistra.io/api/core/v1alpha1"
+
+	"istio.io/istio/pilot/pkg/serviceregistry"
+	"istio.io/istio/pilot/pkg/serviceregistry/aggregate"
+	"istio.io/istio/pilot/pkg/serviceregistry/federation"
+	kubecontroller "istio.io/istio/pkg/kube/controller"
+	memberroll "istio.io/istio/pkg/servicemesh/controller"
+	"istio.io/istio/pkg/servicemesh/federation/common"
+)
+
+const controllerName = "federation-imports-controller"
+
+type Options struct {
+	common.ControllerOptions
+	ServiceController *aggregate.Controller
+}
+
+type Controller struct {
+	*kubecontroller.Controller
+	cs                maistraclient.Interface
+	serviceController *aggregate.Controller
+}
+
+// NewController creates a new ServiceImports controller
+func NewController(opt Options) (*Controller, error) {
+	if err := opt.validate(); err != nil {
+		return nil, fmt.Errorf("invalid Options specified for federation import controller: %s", err)
+	}
+
+	cs, err := maistraclient.NewForConfig(opt.KubeClient.RESTConfig())
+	if err != nil {
+		return nil, fmt.Errorf("error creating ClientSet for ServiceMesh: %v", err)
+	}
+
+	mrc := opt.KubeClient.GetMemberRoll()
+
+	return internalNewController(cs, mrc, opt), nil
+}
+
+// allows using a fake client set for testing purposes
+func internalNewController(cs maistraclient.Interface, mrc memberroll.MemberRollController, opt Options) *Controller {
+	logger := common.Logger.WithLabels("component", controllerName)
+	var informer cache.SharedIndexInformer
+	// Currently, we only watch istio system namespace for MeshFederation resources, which is why this block is disabled.
+	if mrc != nil && false {
```

Comment:
yes/no.  I think it makes sense to use a unique namespace for federation, but we have issues deploying gateways to namespaces outside istio-system, so constraining things to istio-system was the cleanest at this point.

Reply:
I don't understand why this block is here if it's never going to be executed and we don't have plans to enable it.

Reply:
So it doesn't get lost?

Reply:
yeah, that's what I thought. That's why I suggested to create a Jira for it so it doesn't get lost. If it's something we want to tackle before 2.1 final, I'd say put it in the sprint with blocker severity so every sprint we remember it :)
(in order to avoid this https://github.com/maistra/istio-operator/blob/maistra-2.1/build/patch-charts.sh#L449)

Reply:
+1 to adding a ticket